Code/Editor/Plugins/EditorAssetImporter/AssetImporterPlugin.cpp

```diff
@@ -85,20 +98,64 @@ AZStd::unique_ptr<AZ::DynamicModuleHandle> AssetImporterPlugin::LoadSceneLibrary
     }
 }
 
-void AssetImporterPlugin::EditImportSettings(const AZStd::string& sourceFilePath)
+QMainWindow* AssetImporterPlugin::EditImportSettings(const AZStd::string& sourceFilePath)
 {
     const QtViewPane* assetImporterPane = GetIEditor()->OpenView(m_toolName.c_str());
 
     if(!assetImporterPane)
     {
-        return;
+        return nullptr;
     }
 
     AssetImporterWindow* assetImporterWindow = qobject_cast<AssetImporterWindow*>(assetImporterPane->Widget());
     if (!assetImporterWindow)
     {
-        return;
+        return nullptr;
     }
 
     assetImporterWindow->OpenFile(sourceFilePath);
+    return assetImporterWindow;
+}
+
+SceneSettingsAssetImporterForPythonRequestHandler::SceneSettingsAssetImporterForPythonRequestHandler()
+{
+    SceneSettingsAssetImporterForPythonRequestBus::Handler::BusConnect();
+}
+
+SceneSettingsAssetImporterForPythonRequestHandler ::~SceneSettingsAssetImporterForPythonRequestHandler()
+{
+    SceneSettingsAssetImporterForPythonRequestBus::Handler::BusDisconnect();
+}
+
+void SceneSettingsAssetImporterForPythonRequestHandler::Reflect(AZ::ReflectContext* context)
+{
+    if (auto* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+    {
+        serialize->Class<SceneSettingsAssetImporterForPythonRequestHandler>()->Version(0);
+    }
+
+    if (AZ::BehaviorContext* behavior = azrtti_cast<AZ::BehaviorContext*>(context))
+    {
+        behavior->EBus<SceneSettingsAssetImporterForPythonRequestBus>("SceneSettingsAssetImporterForPythonRequestBus")
+            ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+            ->Attribute(AZ::Script::Attributes::Module, "qt")
+            ->Event("EditImportSettings", &SceneSettingsAssetImporterForPythonRequestBus::Events::EditImportSettings);
+    }
+}
+
+AZ::u64 SceneSettingsAssetImporterForPythonRequestHandler::EditImportSettings(const AZStd::string& sourceFilePath)
+{
+    QMainWindow* importSettingsWindow = AssetImporterPlugin::GetInstance()->EditImportSettings(sourceFilePath);
+
+    if (!importSettingsWindow)
+    {
+        // There doesn't seem to be a defined invalid Qt WId, so this matches
+        // the behavior of the QtForPythonSystemComponent.cpp, which uses a window ID of 0 if it
+        // can't find the real window ID.
+        return 0;
```

Comment:
Nit: Should we throw any error or warning in this case?

Reply:
The test will fail if this occurs, it won't be able to find the window later if it's given that ID of 0, or if it gets a window it won't find the widgets it expects.